src/main.rs

```diff
@@ -86,19 +77,18 @@ fn main() {
                     if hand.rolls_left == 0 {
                         println!("No rolls left");
                     } else {
-                        let position_to_reroll: Vec<_> = words[1..].iter().collect();
-                        let position_to_reroll: Vec<_> = position_to_reroll
+                        let mut reroll_flags = vec![false; hand.dice.len()];
```

Comment:
I think this way (modifying a flag vector instead) is cleaner, but its really just personal preference. On another note, you can avoid needing a default value with `unwrap_or` by using `flat_map`, which just takes all the `Some` values when used with `Option`.